src/nidcpower/system_tests/test_system_nidcpower.py

```diff
@@ -1063,6 +1065,50 @@ def test_get_lcr_compensation_last_date_and_time(self, session, compensation_typ
         assert last_compensation_datetime.hour == 0
         assert last_compensation_datetime.minute == 0
 
+    # Multi-Threading tests
+    def test_multi_threading_lock_unlock(self, session):
+        release_lock = threading.Event()
+
+        def lock_wait_unlock():
+            session.lock()
+            release_lock.wait()
+            session.unlock()
+
+        def lock_unlock():
+            session.lock()
+            session.unlock()
+
+        # test that lock, unlock functions work properly
+        # No need to test locking for driver, but the gRPC_interpeter doesn't use the driver to lock
```

Comment:
As written, I don't think this sentence adds that much value. Are you saying something like:

"We don't need to test locking on the driver because thats's out of scope for test coverage in nimi-python, but we do need to test the `gRPC_interpeter` (is that the exact name?) because it implements locking in Python" ?

You are still testing locking on the driver, aren't you?

Reply:
Yes, that was what I was saying. But you're right, I'm testing it anyway, so the sentence doesn't add much value.
The correct name is GrpcStubInterpreter. I should use the right name in our comments.

I'll delete this comment.

Reply:
Comment deleted.